indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -609,7 +609,7 @@ private URL makeWorkerURL(Worker worker, String path)
     Preconditions.checkArgument(path.startsWith("/"), "path must start with '/': %s", path);
 
     try {
-      return new URL(StringUtils.format("%s://%s/druid/worker/v1%s", worker.getScheme(), worker.getHost(), path));
+      return new URL(StringUtils.format("%s://%s/druid/worker/v1%s", worker.getScheme() == null ? "http": worker.getScheme(), worker.getHost(), path));
```

Comment:
can you update Worker class instead to handle null scheme in constructor so that its covered for all the places worker.getScheme() would have been used?

Reply:
I think handling it in the Worker class makes sense too.

Also that would fix other usages of `worker.getScheme()` that don't have null checks (looks like there is 1 other).

Reply:
changed